internal/observability.go

```diff
@@ -28,80 +28,86 @@ import (
 var AllViews = []*view.View{
-	ViewReceivedSpansReceiver,
-	ViewExportedSpans,
+	ViewReceiverReceivedSpans,
+	ViewReceiverDroppedSpans,
+	ViewExporterReceivedSpans,
+	ViewExporterDroppedSpans,
 }
 
-// ContextWithReceiverName adds the tag "opencensus_receiver" and the name of the
+// ContextWithTraceReceiverName adds the tag "opencensus_trace_receiver" and the name of the
 // receiver as the value, and returns the newly created context.
-func ContextWithReceiverName(ctx context.Context, receiverName string) context.Context {
-	ctx, _ = tag.New(ctx, tag.Upsert(tagKeyReceiverName, receiverName))
+func ContextWithTraceReceiverName(ctx context.Context, traceReceiverName string) context.Context {
+	ctx, _ = tag.New(ctx, tag.Upsert(tagKeyTraceReceiverName, traceReceiverName))
 	return ctx
 }
 
-// NewReceivedSpansRecorderStreaming creates a function that uses a context created
-// from the name of the receiver to record the number of the spans received
-// by the receiver.
-func NewReceivedSpansRecorderStreaming(lifetimeCtx context.Context, receiverName string) func(*commonpb.Node, []*tracepb.Span) {
-	// We create and reuse this context because for streaming RPCs e.g. with gRPC
-	// the context doesn't change, so it is more useful for avoid expensively adding
-	// keys on each invocation. We can create the context once and then reuse it
-	// when recording measurements.
-	ctx := ContextWithReceiverName(lifetimeCtx, receiverName)
-
-	return func(ni *commonpb.Node, spans []*tracepb.Span) {
-		// TODO: (@odeke-em) perhaps also record information from the node?
-		stats.Record(ctx, mReceivedSpans.M(int64(len(spans))))
-	}
+// RecordTraceReceiverMetrics records the number of the spans received and dropped by the receiver.
+// Use it with a context.Context generated using ContextWithTraceReceiverName().
+func RecordTraceReceiverMetrics(ctxWithTraceReceiverName context.Context, receivedSpans int, droppedSpans int) {
+	stats.Record(ctxWithTraceReceiverName, mReceiverReceivedSpans.M(int64(receivedSpans)), mReceiverDroppedSpans.M(int64(droppedSpans)))
 }
 
-// NewExportedSpansRecorder creates a helper function that'll add the name of the
-// creating exporter as a tag value in the context that will be used to count the
-// the number of spans exported.
-func NewExportedSpansRecorder(exporterName string) func(context.Context, *commonpb.Node, []*tracepb.Span) {
-	return func(ctx context.Context, ni *commonpb.Node, spans []*tracepb.Span) {
-		ctx, _ = tag.New(ctx, tag.Upsert(tagKeyExporterName, exporterName))
-		stats.Record(ctx, mExportedSpans.M(int64(len(spans))))
-	}
+// ContextWithTraceExporterName adds the tag "opencensus_trace_exporter" and the name of the
+// receiver as the value, and returns the newly created context.
+func ContextWithTraceExporterName(ctx context.Context, traceReceiverName string) context.Context {
+	ctx, _ = tag.New(ctx, tag.Upsert(tagKeyTraceReceiverName, traceReceiverName))
```

Comment:
fix tag name and parameter name.

Reply:
thanks. I will add e2e tests in a separate PR.